tests/config/basic/butler.yaml

```diff
@@ -3,6 +3,10 @@ datastore:
   cls: lsst.daf.butler.datastores.posixDatastore.PosixDatastore
   root: ./butler_test_repository
   create: true
+  templates:
+    default: "{datasettype}/{tract:?}/{patch:?}/{filter:?}/{visit:?}"
+    calexp: "{datasettype}.{component:?}/{datasettype}_v{visit}_f{filter}_{component:?}"
+    metric: "{datasettype}.{component:?}/{datasettype}_v{visit:08d}_f{filter}_{component:?}"
```

Comment:
Per `DatasetType` templates should (probably) live in `Registry`. Given that we are dynamically adding them.

Reply:
They are in datastore because we explicitly decided they should be moved inside datastore (before we had storage hints which would have been the butler using the templates). Not all datastores should define templates so I'm not sure why registry wants to define them. Is the issue that DataSetTypes are defined in registry and it seems odd that data store needs to have its own list of them? If I have chained datastores shouldn't I be able to use different templates in each one?